File: src/main/scala/com/lp/test/serialization/KafkaEventSchema.scala
package com.lp.test.serialization

import net.sf.json.JSONObject
import org.apache.flink.api.common.serialization.{DeserializationSchema, SerializationSchema}
import org.apache.flink.api.common.typeinfo.TypeInformation

/**
  * <p/> 
  * <li>Description: 自定义Json反序列化器</li>
  * <li>@author: lipan@cechealth.cn</li> 
  * <li>Date: 2019-05-08 22:06</li> 
  */
class KafkaEventSchema extends DeserializationSchema[JSONObject] with SerializationSchema[JSONObject]{

  override def deserialize(message: Array[Byte]): JSONObject = {
    JSONObject.fromObject(new String(message))
  }

  override def isEndOfStream(nextElement: JSONObject): Boolean = false

  override def serialize(element: JSONObject): Array[Byte] = {
    element.toString().getBytes()
  }

  override def getProducedType: TypeInformation[JSONObject] = {
    TypeInformation.of(classOf[JSONObject])
  }
}

File: src/main/java/com/lp/java/demo/datastream/watermark/KafkaEventSchema.java
package com.lp.java.demo.datastream.watermark;

import net.sf.json.JSONObject;
import org.apache.flink.api.common.serialization.DeserializationSchema;
import org.apache.flink.api.common.serialization.SerializationSchema;
import org.apache.flink.api.common.typeinfo.TypeInformation;

import java.io.IOException;

/**
 * <p/>
 * <li>title: 自定义Json反序列化器</li>
 * <li>@author: li.pan</li>
 * <li>Date: 2020/01/07 22:07 下午</li>
 * <li>Version: V1.0</li>
 * <li>Description: 自定义元素个数触发器</li>
 */
public class KafkaEventSchema implements DeserializationSchema<JSONObject>, SerializationSchema<JSONObject> {

    private static final long serialVersionUID = 6154188370181669758L;

    @Override
    public byte[] serialize(JSONObject event) {
        return event.toString().getBytes();
    }

    @Override
    public JSONObject deserialize(byte[] message) throws IOException {
        return JSONObject.fromObject(new String(message));
    }

    @Override
    public boolean isEndOfStream(JSONObject nextElement) {
        return false;
    }

    @Override
    public TypeInformation<JSONObject> getProducedType() {
        return TypeInformation.of(JSONObject.class);
    }
}